### ISSUE TITLE
Menus API: handle localized API errors

### ISSUE THREAD

**Issue by mzorz** [wordpress-mobile/WordPress-Android#4180]
### Expected behavior

The app should interpret the error object returned by the API and show a nice error to the user. Also, the app should send `locale` parameter to Menus API so we can obtain localized errors

### Actual behavior

We're only showing generic errors without telling the user what actually happened server-side, and this information is coming through in the error object

### Steps to reproduce the behavior
For example, try to create a menu with the exact name of an existing menu. You'll get a  "Could not create menu"  toast, but the relevant error object coming from the server shows:

```
{
	"error": "menu_exists",
	"message": "The menu name <strong>Menu 11<\/strong> conflicts with another menu name. Please try another."
}
```

**Comment by nbradbury** [comment]
@mzorz The mis-named [errStringFromVolleyError](https://github.com/wordpress-mobile/WordPress-Android/blob/develop/WordPress/src/main/java/org/wordpress/android/util/VolleyUtils.java#L27) may be of use here.

**Comment by mzorz** [comment]
actually I think I want to use [messageStringFromVolleyError](https://github.com/wordpress-mobile/WordPress-Android/blob/develop/WordPress/src/main/java/org/wordpress/android/util/VolleyUtils.java#L49) - thanks a lot for hinting on this!

**Comment by nbradbury** [comment]
> actually I think I want to use messageStringFromVolleyError

Whoops, yes - that's right.

**Comment by mzorz** [comment]
Re-check this after merging PR #4190

**Comment by mzorz** [comment]
Closed via #4226
